egs2/librispeech/asr2/local/data.sh

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Xuankai Chang (Carnegie Mellon University)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+
+stage=1
+stop_stage=100000
+skip_stages=         # Spicify the stage to be skipped
+nj=16
+gpu_nj=4
+python=python3       # Specify python to execute espnet commands.
+feats_root=data/extracted_features
+audio_format=flac    # Audio format: wav, flac, wav.ark, flac.ark  (only in feats_type=raw).
+fs=16k               # Sampling rate.
+
+train_set="train_960"
+dev_set="dev"
+test_sets="test_clean test_other dev_clean dev_other"
+
+src_lang="dc"
+src_case="rm"
+tgt_lang="en"
+tgt_case="ts"
+
+# Data related
+# typically this indicates:
+#   ssl_model_type (e.g. hubert_large, wavlm_large) +
+#   layer_idx
+feats_type="wavlm_large/24"
+# Kmeans related
+skip_train_kmeans=false
+portion=0.1
+nclusters=2000
+# Speed perturbation related
+speed_perturb_factors=  # perturbation factors, e.g. "0.9 1.0 1.1" (separated by space).
+
+measure_label_quality=true
+
+
+log "$0 $*"
+. utils/parse_options.sh
+
+. ./db.sh
+. ./path.sh
+. ./cmd.sh
+
+
+if [ $# -ne 0 ]; then
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+model_type=$(echo "${feats_type}" | cut -d/ -f1)
+layer=$(echo "${feats_type}" | cut -d/ -f2)
+
+
+if [ -z "${speed_perturb_factors}" ]; then
+    skip_stages+="2"
```

Comment:
What if the input argument `skip_stages` is not empty and does not end with space?

Reply:
add a space before `2`.